src/uu/cp/src/cp.rs

```diff
@@ -1910,6 +1941,51 @@ fn copy_helper(
     Ok(())
 }
 
+/// Copy the file from `source` to `dest` by reading into and writing from a buffer.
+/// Increment the progress bar on each write to show the intermediate progress
+/// and show any errors that will occur because this is a large file.
+pub fn copy_with_intermediate_progress(
+    source: &Path,
+    dest: &Path,
+    progress_bar: &Option<ProgressBar>,
+) -> CopyResult<()> {
+    let metadata = fs::metadata(source)?;
+    let source_file = OpenOptions::new().read(true).open(source)?;
+    let dest_file = OpenOptions::new()
+        .append(true)
+        .write(true)
+        .create_new(true)
+        .open(dest)
+        .unwrap();
+    let source_size = metadata.len() as usize;
+    progress_bar.as_ref().unwrap().tick();
+    let mut written = 0;
+    while written < source_size {
```

Comment:
Could it be written like ?

while written < source_size {
        let mut buffer = [0; BUFFER_SIZE];
        let read = source_file.read_at(&mut buffer, written as u64)?;
        if read == 0 {
            return Err("No bytes to read".into());
        }
        
        let bytes_written = dest_file.write_at(&buffer[..read], written as u64)?;
        written += bytes_written;

Reply:
Thanks! Looks more readable now. Done!